src/library/commissioner_safe.cpp

```diff
@@ -76,13 +76,16 @@ Error CommissionerSafe::Init(const Config &aConfig)
     VerifyOrExit(event_assign(&mInvokeEvent, mEventBase.Get(), -1, EV_PERSIST, Invoke, this) == 0);
     VerifyOrExit(event_add(&mInvokeEvent, nullptr) == 0);
 
+    // Do not propogate the returned error.
+    SuccessOrExit(StartEventLoopThread());
```

Comment:
Could creating thread ever fail ? If it failed, can we just return Success?

Reply:
@simonlingoogle 
`StartEventLoopThread` will fail only if There has already been an "event loop Thread". We should avoid ignoring error like this. What about making `StartEventLoopThread` returns `void` and use `VerifyOrDie` inside it to make sure the developer is not using it wrong ( `StartEventLoopThread` should be only called when a new commissioner is created).

Reply:
> `StartEventLoopThread` will fail only if There has already been an "event loop Thread". 

Not very familiar with C++ `std::thread`, however I think it is always possible to unable to create a new thread from OS's perspective (maybe due to resource limits). I think in these situations, thread constructor will just fail with exceptions. Do you think it's necessary to convert this exception to `Error`?

Reply:
Yes, you are. The overall strategy of handling exceptions from STL is not handling it and let it die when an exception is thrown. Just like we don't catch exceptions on `std::make_shared<...>(...)`.